source/module_hsolver/diago_cusolver.cpp

```diff
@@ -117,15 +117,15 @@ namespace hsolver
         hamilt::MatrixBlock<T> h_mat_g, s_mat_g;
 
         // global psi for distribute
-        T* psi_g;
+        T* psi_g = nullptr;
 
         // get the context and process information
         int ctxt = ParaV->blacs_ctxt;
         int nprows, npcols, myprow, mypcol;
```

Comment:
one variable for only one line, then initialize the variables.

Reply:
fixed